contracts/JBTiered721Delegate.sol

```diff
@@ -1,776 +1,632 @@
+    function initialize(
+        uint256 _projectId,
+        IJBDirectory _directory,
+        string memory _name,
+        string memory _symbol,
+        IJBFundingCycleStore _fundingCycleStore,
+        string memory _baseUri,
+        IJB721TokenUriResolver _tokenUriResolver,
+        string memory _contractUri,
+        JB721PricingParams memory _pricing,
+        IJBTiered721DelegateStore _store,
+        JBTiered721Flags memory _flags
+    ) public override {
+        // Make the original un-initializable.
+        if (address(this) == codeOrigin) revert();
+
+        // Stop re-initialization.
+        if (address(store) != address(0)) revert();
+
+        // Initialize the superclass.
+        JB721Delegate._initialize(_projectId, _directory, _name, _symbol);
+
+        fundingCycleStore = _fundingCycleStore;
+        store = _store;
+
+        uint256 _packed;
+        // currency in bits 0-47 (48 bits).
+        _packed |= uint256(_pricing.currency);
+        // decimals in bits 48-95 (48 bits).
+        _packed |= uint256(_pricing.decimals) << 48;
+        // prices in bits 96-255 (160 bits).
+        _packed |= uint256(uint160(address(_pricing.prices))) << 96;
+        // Store the packed value.
+        _packedPricingContext = _packed;
 
-  /** 
-    @notice
-    Mint reserved tokens within the tier for the provided value.
+        // Store the base URI if provided.
+        if (bytes(_baseUri).length != 0) baseURI = _baseUri;
 
-    @param _tierId The ID of the tier to mint within.
-    @param _count The number of reserved tokens to mint. 
-  */
-  function mintReservesFor(uint256 _tierId, uint256 _count) public override {
-    // Get a reference to the project's current funding cycle.
-    JBFundingCycle memory _fundingCycle = fundingCycleStore.currentOf(projectId);
+        // Set the contract URI if provided.
+        if (bytes(_contractUri).length != 0) contractURI = _contractUri;
 
-    // Minting reserves must not be paused.
-    if (
-      JBTiered721FundingCycleMetadataResolver.mintingReservesPaused(
-        (JBFundingCycleMetadataResolver.metadata(_fundingCycle))
-      )
-    ) revert RESERVED_TOKEN_MINTING_PAUSED();
+        // Set the token URI resolver if provided.
+        if (_tokenUriResolver != IJB721TokenUriResolver(address(0))) {
+            _store.recordSetTokenUriResolver(_tokenUriResolver);
+        }
 
-    // Record the minted reserves for the tier.
-    uint256[] memory _tokenIds = store.recordMintReservesFor(_tierId, _count);
+        // Record adding the provided tiers.
+        if (_pricing.tiers.length != 0) _store.recordAddTiers(_pricing.tiers);
 
-    // Keep a reference to the reserved token beneficiary.
-    address _reservedTokenBeneficiary = store.reservedTokenBeneficiaryOf(address(this), _tierId);
+        // Set the flags if needed.
+        if (
+            _flags.lockReservedTokenChanges || _flags.lockVotingUnitChanges || _flags.lockManualMintingChanges
```

Comment:
If we want to golf a bit, we could just mload the struct (as it's in mem) then check if not zero, but maybe overshooting